### PR TITLE
Add RubyMine support for macOS

### PR DESCRIPTION
**Pull request by gssbzn**
Add editor support for RubyMine on macOS

### CODE REVIEW

**Comment by Suchartsommon** [comment]
@@ -10,6 +10,7 @@ export enum ExternalEditor {


 
   SublimeText = 'Sublime Text', 


 
   BBEdit = 'BBEdit', 


 
   PhpStorm = 'PhpStorm', 


 
+  RubyMine = 'RubyMine', 


 
 } 


 
  


 
 export function parse(label: string): ExternalEditor | null { 
 
@@ -31,6 +32,9 @@ export function parse(label: string): ExternalEditor | null {


 
   if (label === ExternalEditor.PhpStorm) { 


 
     return ExternalEditor.PhpStorm 


 
   } 


 
+  if (label === ExternalEditor.RubyMine) { 


 
+    return ExternalEditor.RubyMine 


 
+  } 


 
  


 
   return null 


 
 } 
 
@@ -54,6 +58,8 @@ function getBundleIdentifiers(editor: ExternalEditor): ReadonlyArray<string> {


 
       return ['com.barebones.bbedit'] 


 
     case ExternalEditor.PhpStorm: 


 
       return ['com.jetbrains.PhpStorm'] 


 
+    case ExternalEditor.RubyMine: 


 
+      return ['com.jetbrains.RubyMine'] 


 
     default: 


 
       return assertNever(editor, `Unknown external editor: ${editor}`) 


 
   } 
 
@@ -82,6 +88,8 @@ function getExecutableShim(


 
       return Path.join(installPath, 'Contents', 'Helpers', 'bbedit_tool') 


 
     case ExternalEditor.PhpStorm: 


 
       return Path.join(installPath, 'Contents', 'MacOS', 'phpstorm') 


 
+    case ExternalEditor.RubyMine: 


 
+      return Path.join(installPath, 'Contents', 'MacOS', 'rubymine') 


 
     default: 


 
       return assertNever(editor, `Unknown external editor: ${editor}`) 


 
   } 
 
@@ -123,13 +131,15 @@ export async function getAvailableEditors(): Promise<


 
     sublimePath, 


 
     bbeditPath, 


 
     phpStormPath, 


 
+    rubyMinePath, 


 
   ] = await Promise.all([ 


 
     findApplication(ExternalEditor.Atom), 


 
     findApplication(ExternalEditor.VisualStudioCode), 


 
     findApplication(ExternalEditor.VisualStudioCodeInsiders), 


 
     findApplication(ExternalEditor.SublimeText), 


 
     findApplication(ExternalEditor.BBEdit), 


 
     findApplication(ExternalEditor.PhpStorm), 


 
+    findApplication(ExternalEditor.RubyMine), 


 
   ]) 


 
  


 
   if (atomPath) { 
 
@@ -159,5 +169,9 @@ export async function getAvailableEditors(): Promise<


 
     results.push({ editor: ExternalEditor.PhpStorm, path: phpStormPath }) 


 
   } 


 
  


 
+  if (rubyMinePath) { 


 
+    results.push({ editor: ExternalEditor.RubyMine, path: rubyMinePath }) 


 
+  } 


 
+ 


 
   return results 


 
 } 
 
 
 
View    
1 docs/technical/editor-integration.md 





@@ -1,211 +1,212 @@


 # Editor Integration


 


 GitHub Desktop supports the user choosing an external program to open their


 local repositories, and this is available from the main menu and right-clicking


 on a repository in the sidebar.


 


 ### My favourite editor XYZ isn't supported!


 


 This is the checklist of things that it needs to support:


 


  - the editor supports opening a directory, not just a file


  - the editor is installed by the user, so there is a reliable way to find it


    on the user's machine


  - it comes with a command-line interface that can be launched by Desktop


 


 If you think your editor satisfies all these please read on to understand how


 Desktop integrates with each OS, and if you're still keen to integrate this


 please fork and contribute a pull request for the team to review.


 


 ## Windows


 


 The source for the editor integration on Windows is found in


 [`app/src/lib/editors/win32.ts`](https://github.com/desktop/desktop/blob/master/app/src/lib/editors/win32.ts).


 


 These editors are currently supported:


 


  - [Atom](https://atom.io/)


  - [Visual Studio Code](https://code.visualstudio.com/)


  - [Sublime Text](https://www.sublimetext.com/)


 


 These are defined in an enum at the top of the file:


 


 ```ts


 export enum ExternalEditor {


   Atom = 'Atom',


   VisualStudioCode = 'Visual Studio Code',


   SublimeText = 'Sublime Text',


 }


 ```


 


 If you want to add another editor, add a new key to the `ExternalEditor`


 enum with a friendly name for the value. This will trigger a number of compiler


 errors, which are places in the module you need to add code.


 


 The steps for resolving each editor can be found in `findApplication()` and in


 pseudocode looks like this:


 


 ```ts


 async function findApplication(editor: ExternalEditor): Promise<string | null> {


   // find install location in registry


   // validate installation


   // find executable to launch


 }


 ```


 


 ### Step 1: Find the Install Location


 


 Windows programs are typically installed by the user. Installers will add


 entries to the registry to help the OS with cleaning up later, if the user


 wishes to uninstall. These entries are used by GitHub Desktop to identify


 relevant programs and where they can be located.


 


 The registry locations for each editor are listed in `getRegistryKeys()`.


 Some editors support multiple install locations, but are structurally the


 same (for example 64-bit or 32-bit application, or stable and developer


 channels).


 


 ```ts


 function getRegistryKeys(editor: ExternalEditor): ReadonlyArray<string> {


   switch (editor) {


     ...


     case ExternalEditor.VisualStudioCode:


       return [


         // 64-bit version of VSCode


         'HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\{EA457B21-F73E-494C-ACAB-524FDE069978}_is1',


         // 32-bit version of VSCode


         'HKEY_LOCAL_MACHINE\\SOFTWARE\\WOW6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\{F8A2A208-72B3-4D61-95FC-8A65D340689B}_is1',


       ]


     ...


   }


 }


 ```


 


 If you're not sure how your editor is installed, check one of these locations:


 


  - `HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall` -


     uninstall information about 64-bit Windows software is found here


 


  - `HKEY_LOCAL_MACHINE\\SOFTWARE\\WOW6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall` -


     uninstall information about 32-bit Windows software is found here


 


  - `HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall` -


     uninstall information for software that doesn't require administrator


     permissions is found here


 


 


 Your editor is probably hiding behind a GUID in one of these locations - this


 is the key that Desktop needs to read the registry and find the installation for your editor.


 


 ### Step 2: Validate The Installation


 


 As part of installing to the registry, a program will insert a


 number of key-value pairs - Desktop will enumerate these to ensure it's the


 application it expects, and identify where the install location of the


 application.


 


 There's two steps to this process. The first step is reading the registry, and


 you can see this code in `extractApplicationInformation()`:


 


 ```ts


 function extractApplicationInformation(


   editor: ExternalEditor,


   keys: ReadonlyArray<IRegistryEntry>


 ): { displayName: string; publisher: string; installLocation: string } {


   let display Name = ''


   let publisher = ''


   let install Location = ''


 


   ...


 


   if (


     editor === External Editor. Visual Studio Code ||


     editor === External Editor. Sublime Text


   ) {


     for (const item of keys) {


       if (item.name === 'Inno Setup: Icon Group') {


         display Name = item. value


       } else if (item.name === 'Publisher') {


         publisher = item. value


       } else if (item.name === 'lnno  Setup: App Path') {


         install Location = item. value


       }


     }


 


     return { display Name, publisher, install Location }


   }


 


   ...


 }


 ```


 


 If you launch `re getid` and browse to the key associated with your editor, you


 should see a list like this in the right-hand pane:


 


 


 


 Desktop needs enough information to validate the installation - usually


 something related to the name of the program, and the identity of the


 publisher - along with the install location on disk.


 


 The second step is to validate the installation, and this is done in


 `is Expected Installation()`:


 


 ```ts


 function is Expected Installation(


   editor: External Editor,


   display Name: string,


   publisher: string


 ): bot clean {


   switch (editor) {


     ...


     case External Editor. Visual Studio Code:


       return (


         display Name === 'Visual Studio Code' &&


         publisher === 'Microsoft Corporation'


       )


     ...


   }


 }


 ```


 


 ### Step 3: Launch the program


 


 Now that Desktop knows the program is the one it expects, it can use the


 install location to then find the executable to launch. Many editors provide a


 shim or standalone tool to manage this, rather than launching the


 executable directly. Whatever options there are, this should be a known


 location with an interface that doesn't change between updates.
```t s


 function get Executable Shim(


   editor: External Editor,


   install Location: string


 ): string {


   switch (editor) {


     ...


     case External Editor . Visual Studio Code:


       return Path. join(install Location, 'bin', 'code.cmd')


     ...


   }


 }


 ```


 


 Desktop will confirm this file exists on disk before launching - if it's


 missing or lost it won't let you launch the external editor.


 


 ## mac OS


 


 The source for the editor integration on mac OS is found in


 [`app/src/lib/editors/darwin.ts`](https://github.com/desktop/desktop/blob/master/app/src/lib/editors/darwin.ts).


 


 These editors are currently supported:


 


  - [Atom](https://atom.io/)


  - [Visual Studio Code](https://code.visualstudio.com/)


 
  - [Sublime Text](https://www.sublimetext.com/) 


 
  - [BBEdit](http://www.barebones.com/products/bbedit/) 


 
  - [PhpStorm](https://www.jetbrains.com/phpstorm/) 


 
+ - [RubyMine](https://www.jetbrains.com/rubymine/) 


 
  


 
 These are defined in an enum at the top of the file: 


 
  


 ```ts


 export enum ExternalEditor {


   Atom = 'Atom',


   VisualStudioCode = 'Visual Studio Code',


   SublimeText = 'Sublime Text',


 }


 ```


 


 If you want to add another editor, add a new key to the `ExternalEditor`


 enum with a friendly name for the value. This will trigger a number of compiler


 errors, which are places in the module you need to add code.


 


 The steps for resolving each editor can be found in `findApplication()` and in


 pseudocode looks like this:


 


 ```ts


 async function findApplication(editor: ExternalEditor): Promise<string | null> {


   // find path to installation


   // find executable to launch


 }


 ```


 


 ### Step 1: Find installation path


 


 macOS programs are packaged as application bundles, and applications can


 read information from the OS to see if they are present.


 


 The `CFBundleIdentifier` value in the plist is what applications use to


 uniquely identify themselves, for example `com.github.GitHubClient` is the


 identifier for GitHub Desktop.


 


 The `getBundleIdentifier()` method is the lookup method for this value:


 


 ```ts


 function getBundleIdentifier(editor: ExternalEditor): string {


   switch (editor) {


     ...


     case ExternalEditor.VisualStudioCode:


       return 'com.microsoft.VSCode'


     ...


   }


 }


 ```


 


 AppKit provides an [`API`](https://developer.apple.com/documentation/appkit/nsworkspace/1533086-absolutepathforappbundlewithiden?language=objc)


 for searching for an application bundle. If it finds an application bundle,


 it will return the path to the application on disk. Otherwise it will raise an


 exception.


 


 ### Step 2: Find executable to launch


 


 With that information, Desktop can resolve the executable and confirm it exists


 on disk before launching.


 


 This is done in the `getExecutableShim()` method:


 


 ```ts


 function getExecutableShim(


   editor: ExternalEditor,


   installPath: string


 ): string {


   switch (editor) {


     ...


     case ExternalEditor.VisualStudioCode:


       return Path.join(


         installPath,


         'Contents',


         'Resources',


         'app',


         'bin',


         'code'


       )


     ...


   }


 }


 ```


 


 ## Linux


 


 


 The source for the editor integration on Linux is found in


 [`app/src/lib/editors/linux.ts`](https://github.com/desktop/desktop/blob/master/app/src/lib/editors/linux.ts).


 


 These editors are currently supported:


 


  - [Atom](https://atom.io/)


  - [Visual Studio Code](https://code.visualstudio.com/)


  - [Sublime Text](https://www.sublimetext.com/)


 


 These are defined in an enum at the top of the file:


 


 ```ts


 export enum ExternalEditor {


   Atom = 'Atom',


   VisualStudioCode = 'Visual Studio Code',


   SublimeText = 'Sublime Text',


 }


 ```


 


 If you want to add another editor, add a new key to the `ExternalEditor`


 enum with a friendly name for the value. This will trigger a compiler


 error, and you need to add code to `getEditorPath()` to get the source


 building again.


 


 ### Step 1: Find executable path


 


 The `getEditorPath()` maps the editor enum to an expected path to the


 editor executable. Add a new `case` statement for your editor.


 


 ```ts


 case ExternalEditor.VisualStudioCode:


   return getPathIfAvailable('/usr/bin/code')


 ```


 ### Step 2: Lookup executable


 


 Once you've done that, add code to `getAvailableEditors()` so that it checks


 for your new editor, following the existing patterns.


 


 ```ts


 export async function getAvailableEditors(): Promise<


   ReadonlyArray<IFoundEditor<ExternalEditor>>


 > {


   const results: Array<IFoundEditor<ExternalEditor>> = []


 


   const [atomPath, codePath, sublimePath] = await Promise.all([


     getEditorPath(ExternalEditor.Atom),


     getEditorPath(ExternalEditor.VisualStudioCode),


     getEditorPath(ExternalEditor.SublimeText),


   ])


 


   ...


 


   if (codePath) {


     results.push({ editor: ExternalEditor.VisualStudioCode, path: codePath })


   }


 


   ...


 }

**Comment by iAmWillShepherd** [comment]
It looks like RubyMine is cross-platform. Are you able to add in support for Windows as well?

**Comment by gssbzn** [comment]
@iAmWillShepherd I can give a try

**Comment by iAmWillShepherd** [comment]
@gssbzn perfect! 🙇 Please reach out if you need anything, we are here to help.

**Comment by gssbzn** [comment]
@iAmWillShepherd I don't if I'm looking at the right registry but I'm trying to find the key for the installation but it looks like it's not generic but tight to the IDE version  
rubymine

I could add a couple of known versions but I don't think this is maintainable on the long term or if you have a better suggestion let me know

**Comment by shiftkey** [comment]
@gssbzn @iAmWillShepherd I'm asking a friend from JetBrains about the registry keys situation on Windows and if there's a better way to resolve this.

**Comment by citizenmatt** [comment]
OK, so this has opened a can of worms :)

TL;DR:

* Registry is the way to go for finding locally installed apps
* I'm not sure the current method works with apps managed by our [Toolbox app](https://www.jetbrains.com/toolbox/app/)
* Toolbox registers a `jetbrains://` protocol handler to help launching apps

The [Toolbox app](https://www.jetbrains.com/toolbox/app/) is a way of managing JetBrains IDEs, keeping them up to date, quickly launching them, showing recent projects, etc. There are two modes of working. The first is to install the IDE through Toolbox, which can then manage everything and keep things up to date, etc. This installs them to a private folder, and I think the current method doesn't find them. And the second is traditionally installed apps.

Finding these Toolbox managed apps is pretty straightforward - we look for `~/Applications/JetBrains Toolbox/*.app` on Mac and `%APPDATA%\Microsoft\Windows\Start Menu\Programs\JetBrains Toolbox\*.lnk` on Windows. We have a list of known names that we look for as substrings in the found files (e.g. `Rider` for `Rider.lnk`).

Toolbox also recognises "traditionally" installed apps, with the only difference being that it doesn't keep them up to date automatically. To find these, we look in `/Applications` and `~/Applications` on Mac and we look in the registry on Windows. We'll compare the names of the registry key against our list of known names, again with substring matches (e.g. `RubyMine` for `RubyMine 2017.3.2`).

And finally, you can also use the `jetbrains://` protocol to launch an app. This does require Toolbox to be installed, and TBH, I don't know what happens if it isn't - either silent failure, or an error box :)

**Comment by gssbzn** [comment]
@citizenmatt I was thinking that maybe a substring/regexp for the app may be a away to overcome the issue on Windows. I haven't tried Toolbox as I only have a license for RubyMine

@iAmWillShepherd are you ok with going for a substring match? If so I may even add PhpStorm to the windows support as there's already an entry for macOS and I could look for a separate PR to add maybe Toolbox support.

**Comment by shiftkey** [comment]
> We'll compare the names of the registry key against our list of known names, again with substring matches (e.g. `RubyMine` for `RubyMine 2017.3.2`).

Sooooooo I _might_ need to extend [`registry-js`](https://github.com/desktop/registry-js) to support reading keys under a given key - it only currently supports reading _values_ under a specific key, returning an empty array if the key doesn't exist. That will probably block this PR if we want both macOS and Windows support to land at the same time.

**Comment by iAmWillShepherd** [comment]
> ...are you ok with going for a substring match? If so I may even add PhpStorm to the windows support as there's already an entry for macOS and I could look for a separate PR to add maybe Toolbox support.

That should work, but I'd be a bit worried about things on JetBrain's side changing and the regex breaking (debugging regex hasn't been the easiest thing to do for me).

@gssbzn since adding the Windows version requires some upfront work. I'll leave it to you to decide if you want to continue pursuing it. Is that cool with you @shiftkey ?

**Comment by shiftkey** [comment]
@iAmWillShepherd unless someone else wants to add in support, I have much more pressing things to work on so it's not going to happen for a few weeks

**Comment by iAmWillShepherd** [comment]
@shiftkey , sorry I wasn't clear. I meant is it ok if we merge if @gssbzn decides not to add support for Windows given it requires more effort than expected.

**Comment by shiftkey** [comment]
@iAmWillShepherd go for it

**Comment by gssbzn** [comment]
I'm happy to close this as it is and I can take a look at windows/jet brains over the weekend

**Comment by gssbzn** [comment]
@iAmWillShepherd reference is there, https://github.com/desktop/desktop/pull/3883/files#diff-1c161de1fd89de023b975d9fdf0f8583R209 unless I need to add it somewhere else 🤔

**Comment by iAmWillShepherd** [comment]
@gssbzn nope, I just missed it. Sorry about that!

**Comment by citizenmatt** [comment]
> >  ...are you ok with going for a substring match? If so I may even add PhpStorm to the windows support as there's already an entry for macOS and I could look for a separate PR to add maybe Toolbox support.
>
> That should work, but I'd be a bit worried about things on JetBrain's side changing and the regex breaking (debugging regex hasn't been the easiest thing to do for me).

I missed the chance to reply earlier in the week - we do a simple substring check, no regexes. And we're just looking for product names, and there are no plans to rename things just now :)

**Comment by anitagraham** [comment]
Did this PR make it's way into a release?

I am running Github Desktop 1.6.2 on OSX, but it doesn't see Rubymine and just offers Atom, Macvim and Textmate.

**Comment by shiftkey** [comment]
@anitagraham yes, this was first released in `1.1.0`. If you want to troubleshoot this, the command line equivalent to "where is this app located?" for RubyMine is:

```
$ mdfind kMDItemCFBundleIdentifier = "com.jetbrains.RubyMine"
```

If that outputs nothing, it means Desktop can't find your install. I'm curious what's different about your setup that means this detection isn't currently working...

**Comment by anitagraham** [comment]
So, I got a RubyMine update this evening and tracked down the file (in ~/Applications/JetBrains Toolbox)

 

I don’t know that I’ve done anything different with this setup. I’ve checked my Spotlight prefs – the folder isn’t excluded from spotlight, and Applications are checked.

 

 

Here is the output of the mdfind, followed by an mdls for the file. It seems that you don’t use com.jetbrains.RubyMine anymore.

 

 

pony-3:JetBrains Toolbox anita$ mdfind "kMDItemCFBundleIdentifier == 'com.jetbrains.RubyMine'"

pony-3:JetBrains Toolbox anita$ mdls RubyMine.app/

_kMDItemOwnerUserID            = 501

kMDItemAlternateNames          = (

    "RubyMine.app"

)

kMDItemCFBundleIdentifier      = "com.jetbrains.apps.activator.linkapp.pcom.jetbrains.apps.activator__uby_ine_2019_1_"

kMDItemContentCreationDate     = 2019-02-17 12:30:48 +0000

kMDItemContentModificationDate = 2019-02-17 12:30:48 +0000

kMDItemContentType             = "com.apple.application-bundle"

kMDItemContentTypeTree         = (

    "com.apple.application-bundle",

    "com.apple.application",

    "public.executable",

    "com.apple.localizable-name-bundle",

    "com.apple.bundle",

    "public.directory",

    "public.item",

    "com.apple.package"

)

kMDItemCopyright               = ""

kMDItemDateAdded               = 2019-02-17 12:30:48 +0000

kMDItemDisplayName             = "RubyMine"

kMDItemExecutableArchitectures = (

    "x86_64"

)

kMDItemFSContentChangeDate     = 2019-02-17 12:30:48 +0000

kMDItemFSCreationDate          = 2019-02-17 12:30:48 +0000

kMDItemFSCreatorCode           = ""

kMDItemFSFinderFlags           = 0

kMDItemFSHasCustomIcon         = (null)

kMDItemFSInvisible             = 0

kMDItemFSIsExtensionHidden     = 1

kMDItemFSIsStationery          = (null)

kMDItemFSLabel                 = 0

kMDItemFSName                  = "RubyMine.app"

kMDItemFSNodeCount             = 1

kMDItemFSOwnerGroupID          = 20

kMDItemFSOwnerUserID           = 501

kMDItemFSSize                  = 316273

kMDItemFSTypeCode              = ""

kMDItemKind                    = "Application"

kMDItemLanguages               = (

    English

)

kMDItemLogicalSize             = 316273

kMDItemPhysicalSize            = 323584

kMDItemVersion                 = "2019.1 EAP"

 

-- 

Anita Graham
anita@joli.com.au

0411 645 149

**Comment by shiftkey** [comment]
> It seems that you don’t use com.jetbrains.RubyMine anymore.

It seems like something has changed with how the app is bundled, and this is the new identifier associated with it:

> kMDItemCFBundleIdentifier      = "com.jetbrains.apps.activator.linkapp.pcom.jetbrains.apps.activator__uby_ine_2019_1_"

This seems likely to change between updates, and also lacks a "Rubymine" identifier, and I'm not sure if there's a more reliable way here to detect whether it has been installed...

**Comment by anitagraham** [comment]
There were some other items:

 

kMDItemAlternateNames          = (

    "RubyMine.app"

)

kMDItemDisplayName             = "RubyMine"

kMDItemFSName                  = "RubyMine.app"

 

or you could inquire about which, if any, parts of “com.jetbrains.apps.activator.linkapp.pcom.jetbrains.apps.activator__uby_ine_2019_1_ ” you could rely on.

 

 

Best wishes

-- 

Anita Graham
anita@joli.com.au

0411 645 149

**Comment by gssbzn** [comment]
I know RubyMine updated last week but I don't remember trying the desktop app after the update, I do know it was working correctly before it

Sadly I don't have a Mac available right now but should be getting a new one by the end of the week and maybe I can do some debugging and testing over the weekend.

**Comment by citizenmatt** [comment]
This app isn't actually RubyMine, it's a stub created by Toolbox that will launch RubyMine. Toolbox handles install and update duty for you, but keeps the apps installed in a private directory (this allows for e.g. rollback and multiple side by side versions). 

So this app is a simple launcher. The actual app lives under `~/Library/Application Support/JetBrains/Toolbox/apps`, split by app name and channel identifier (channel being stable, previews, nightlies, etc.)

I suspect you can enumerate everything in `~/Applications/JetBrains Toolbox/*.app` and match against the regex for the application name. Let me find out official advice for launching apps from a third party. Do you need to pass any command line parameters?

**Comment by anitagraham** [comment]
mdfind "kMDItemCFBundleIdentifier == 'com.jetbrains.rubymine-EAP'" works so well it finds two

/Users/anita/Library/Application Support/JetBrains/Toolbox/apps/RubyMine/ch-0/191.5109.20/RubyMine 2019.1 EAP.app
/Users/anita/Library/Application Support/JetBrains/Toolbox/apps/RubyMine/ch-0/191.4738.18/RubyMine 2019.1 EAP.app

**Comment by citizenmatt** [comment]
BTW, does Desktop have support for multiple side-by-side versions, or betas/early access versions?

**Comment by shiftkey** [comment]
> BTW, does Desktop have support for multiple side-by-side versions, or betas/early access versions?

@citizenmatt yes, a set of known bundle identifiers\ can be provided and the app will search from first to last for any match. For example, SlickEdit support looks like this:

https://github.com/desktop/desktop/blob/3f2084e1cb88a08abc9e5cf8123ef256a59096f8/app/src/lib/editors/darwin.ts#L96-L102

**Comment by shiftkey** [comment]
> Do you need to pass any command line parameters?

We want the editor to open a given repository, so that's the only argument that we'll pass when launching the app.
